host/traceh.c

```diff
@@ -331,18 +331,46 @@ oe_result_t oe_log_set_callback(void* context, oe_log_callback_t callback)
     return OE_UNEXPECTED;
 }
 
+#ifdef _WIN32
+int gettimeofday(struct timeval* tv, struct timezone* tzp)
+{
+    // EPOCH = 10000000 * 60 * 60 * 24 * (365 * (1970 - 1601) + 89)
+    static const uint64_t EPOCH = ((uint64_t)116444736000000000ULL);
+    static FILETIME file_time;
+    static uint64_t time;
+
+    GetSystemTimePreciseAsFileTime(&file_time);
+    time = ((uint64_t)file_time.dwLowDateTime);
+    time += ((uint64_t)file_time.dwHighDateTime) << 32;
+    time -= EPOCH;
+
+    tv->tv_sec = (long)(time / 10000000L);
+    tv->tv_usec = (long)(time % 10000000L);
+    // Unit conversion, from 100 ns to 1 usec.
+    tv->tv_usec /= 10L;
+    return 0;
+}
+#endif
+
 // This is an expensive operation, it involves acquiring lock
 // and file operation.
 void oe_log_message(bool is_enclave, oe_log_level_t level, const char* message)
 {
     // get timestamp for log
     struct tm t;
     time_t lt = time(NULL);
-    gmtime_r(&lt, &t);
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    lt = tv.tv_sec;
+#ifdef _WIN32
+    localtime_s(&t, &lt);
+#else
+    localtime_r(&lt, &t);
+#endif
 
     char time[20];
-    strftime(time, sizeof(time), "%Y-%m-%dT%H:%M:%S", &t);
-    long int usecs = 0;
+    strftime(time, sizeof(time), "%Y-%m-%d %H:%M:%S", &t);
```

Comment:
If we want to represent local time instead of UTC, I would suggest logging the time zone differential (i.e. use the format string `"%Y-%m-%dT%H:%M:%S%z"`)

The original format string looks like it was trying to be compliant with [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) but it really should have included the 'Z' at the end.

Reply:
Fixed. Thanks, I never noticed the detail.